crates/driver/src/boundary/settlement.rs

```diff
@@ -238,117 +234,109 @@ fn into_boundary_order(order: &competition::auction::Order) -> Order {
+                                    BuyTokenDestination::Erc20
+                                }
+                                competition::order::BuyDestination::Internal => {
+                                    BuyTokenDestination::Internal
+                                }
+                            },
                         },
+                        signature: Default::default(),
                     },
-                    signature: Default::default(),
-                },
-                exec_sell_amount: jit.executed_sell_amount,
-                exec_buy_amount: jit.executed_buy_amount,
+                    exec_sell_amount: jit.order.executed_sell_amount,
+                    exec_buy_amount: jit.order.executed_buy_amount,
+                }),
+                competition::solution::Trade::Fulfillment(_) => None,
             })
             .collect(),
         amms: solution
-            .amms
-            .into_iter()
-            .map(|(address, amms)| {
-                (
-                    address.into(),
+            .interactions
+            .iter()
+            .filter_map(|interaction| match interaction {
+                competition::solution::Interaction::Liquidity(interaction) => Some((
+                    interaction.liquidity.address.into(),
                     UpdatedAmmModel {
-                        execution: amms
-                            .into_iter()
-                            .map(|amm| ExecutedAmmModel {
-                                sell_token: amm.sell.token.into(),
-                                buy_token: amm.buy.token.into(),
-                                exec_sell_amount: amm.sell.amount,
-                                exec_buy_amount: amm.buy.amount,
-                                exec_plan: ExecutionPlan {
-                                    coordinates: ExecutionPlanCoordinatesModel {
-                                        sequence: amm.sequence,
-                                        position: amm.position,
-                                    },
-                                    internal: amm.internal,
-                                },
-                            })
-                            .collect(),
-                        // TODO @nlordell is this right...?
-                        // We're getting rid of this and it's only needed for price estimation, so
-                        // shouldn't be used here?
+                        execution: vec![ExecutedAmmModel {
+                            sell_token: interaction.output.token.into(),
+                            buy_token: interaction.input.token.into(),
+                            exec_sell_amount: interaction.output.amount,
+                            exec_buy_amount: interaction.input.amount,
+                            // TODO I don't know what I should do here
+                            exec_plan: Default::default(),
```

Comment:
The `exec_plan` was used for ordering interactions. You can actually get away with
using `solution.interactions.iter().enumerate()` and then generating the exec plan based on the index of the interaction.

Same for `interaction_data` field 👇.

Reply:
1. In theory, this needs to be part of the domain object and in the `driver` <> `solver/searcher` DTO (I included it here).
2. Yes this is fine. It uses sorting order, so in theory you could use a step of 42 and it would still be OK.

Reply:
> I included it here

I thought I linked it correctly but apparently not:

https://github.com/cowprotocol/services/blob/0f04e6cdd6439c42148abb814a00f3298c0db370/crates/searchers/openapi.yml#L638

Reply:
I think you should probably add `ExecutionPlan` to the `competition::solution::Interaction::Liquidity` and then just map it here (because exec plan is set to be mandatory for amms)

(if I did not completely missed the context of this function `to_boundary_solution`)

Reply:
and the same for `competition::solution::Interaction::Custom`

Reply:
Yes, we are getting rid of them. There is no need for them since the interactions are already ordered in a `Vec` now.